napari/layers/labels/labels.py

```diff
@@ -9,46 +10,97 @@
         Opacity of the labels, must be between 0 and 1.
     name : str, keyword-only
         Name of the layer.
-    num_colors : int, optional
-        Number of unique colors to use. Default used if not given.
-    **kwargs : dict
-        Parameters that will be translated to metadata.
+
+    Attributes
+    ----------
+    data : array
+        Integer valued label data. Can be N dimensional. Every pixel contains
+        an integer ID corresponding to the region it belongs to. The label 0 is
+        rendered as transparent.
+    metadata : dict
+        Labels metadata.
+    num_colors : int
+        Number of unique colors to use in colormap.
+    seed : float
+        Seed for colormap random generator.
+    opacity : float
+        Opacity of the labels, must be between 0 and 1.
+    contiguous : bool
+        If `True`, the fill bucket changes only connected pixels of same label.
+    n_dimensional : bool
+        If `True`, paint and fill edit labels across all dimensions.
+    brush_size : float
+        Size of the paint brush.
+    selected_label : int
+        Index of selected label. Can be greater than the current maximum label.
+    mode : str
+        Interactive mode. The normal, default mode is PAN_ZOOM, which
+        allows for normal interactivity with the canvas.
+
+        In PICKER mode the cursor functions like a color picker, setting the
+        clicked on label to be the curent label. If the background is picked it
+        will select the background label `0`.
+
+        In PAINT mode the cursor functions like a paint brush changing any
+        pixels it brushes over to the current label. If the background label
+        `0` is selected than any pixels will be changed to background and this
+        tool functions like an eraser. The size and shape of the cursor can be
+        adjusted in the properties widget.
+
+        In FILL mode the cursor functions like a fill bucket replacing pixels
+        of the label clicked on with the current label. It can either replace
+        all pixels of that label or just those that are contiguous with the
+        clicked on pixel. If the background label `0` is selected than any
+        pixels will be changed to background and this tool functions like an
+        eraser.
+
+    Extended Summary
+    ----------
+    _data_view : array (N, M)
+        2D labels data for the currently viewed slice.
+    _selected_color : 4-tuple or None
+        RGBA tuple of the color of the selected label, or None if the
+        background label `0` is selected.
+    _last_cursor_coord : list or None
+        Coordinates of last cursor click before painting, gets reset to None
+        after painting is done. Used for interpolating brush strokes.
     """
 
     def __init__(
         self,
-        label_image,
-        meta=None,
-        *,
-        name=None,
+        labels,
+        metadata=None,
         num_colors=50,
+        seed=0.5,
         opacity=0.7,
+        *,
```

Comment:
imho this `*` should be much earlier in the list. The rule of thumb I use is "If I saw a call with just the positional arguments, would I know immediately what they mean?" So, `copyfile('path/to/src', 'path/to/dst')`, sure, but `add_labels(label_image, None, 72, 0.8)` is just terrible. I think it should come immediately after the label_image.

Reply:
agreed, fixed here and for images